dump/mapping/prepbufr_acft_profiles.py

```diff
@@ -0,0 +1,196 @@
+        dhr2 = np.array(dhr)
+        self._replace_timestamp(container, self._get_reference_time(input_path))
+    
+        self.log.debug(f'Make an array of 0s for MetaData/sequenceNumber')
+        sequenceNum = self._compute_sequence_number(lon) 
+        self.log.debug(f'sequenceNum min/max =  {sequenceNum.min()} {sequenceNum.max()}')
+    
+        self.log.debug(f'Compute Obstypes')
+        t_ot = container.get('airTemperatureObservationType')
+        #tv_ot = container.get('virtualTemperatureObservationType')
+        q_ot = container.get('specificHumidityObservationType')
+        uv_ot = container.get('windObservationType')
+        ot_paths = container.get_paths('airTemperatureObservationType')
+    
+        airTemperature = container.get('airTemperatureObsValue')
+        #virtualTemperature = container.get('virtualTemperatureObsValue')
+        specificHumidity = container.get('specificHumidityObsValue')
+        wind = container.get('windNorthwardObsValue')
+    
+        ot_airTemperature = self._compute_typ_other(t_ot, airTemperature)
+        #ot_virtualTemperature = self._compute_typ_other(tv_ot, virtualTemperature)
+        ot_specificHumidity = self._compute_typ_other(q_ot, specificHumidity)
+        ot_wind = self._compute_typ_uv(uv_ot, wind)
+    
+        self.log.debug(f'Change IALR to 0.0 if masked for bias correction.')
+        ialr = container.get('instantaneousAltitudeRate')
+        ialr_paths = container.get_paths('instantaneousAltitudeRate')
+        ialr2 = ma.array(ialr)
+    
+        ialr_bc = self._compute_ialr_if_masked(uv_ot, ialr2)
+    
+        self.log.debug(f'Update variables in container')
+        container.replace('instantaneousAltitudeRate', ialr_bc)#, ot_paths)
+        container.replace('airTemperatureObservationType', ot_airTemperature)
+        #container.replace('virtualTemperatureObservationType', ot_virtualTemperature)
+        container.replace('specificHumidityObservationType', ot_specificHumidity)
+        container.replace('windObservationType', ot_wind)
+    
+        self.log.debug(f'Add variables to container')
+        container.add('sequenceNumber', sequenceNum, lon_paths)
+    
+        # Check
+        self.log.debug(f'container list (updated): {container.list()}')
+    
+        return container
+    
+    
+    def _compute_typ_other(self, typ, var):
+        """
+        Compute datatype if the variable is not wind.
+        Parameters:
+            typ: datatype
+            var: obsValue variable
+        Returns:
+            Masked array of the new datatype
+        """
+    
+        typ_var = copy.deepcopy(typ)
+        typ_var[(typ_var > 300) & (typ_var < 400)] -= 200
+        typ_var[(typ_var > 400) & (typ_var < 500)] -= 300
+        typ_var[(typ_var > 500) & (typ_var < 600)] -= 400
+    
+        for i in range(len(typ_var)):
+            if ma.is_masked(var[i]):
+                typ_var[i] = typ_var.fill_value
+    
```

Comment:
I don't think ma.is_masked works this way. It takes an masked array and tells you if anything is masked. 

You can use the filled() function later to get the filled version of the array. But really I should fix container.replace.

Reply:
This was done to mask values where there was no physical data, but I suppose it doesn't matter. I'll get rid of it.

Reply:
thanks!